File: app/src/main/java/com/example/techfix/data/dao/PaymentDao.java
package com.example.techfix.data.dao;
import androidx.room.*;
import com.example.techfix.data.entity.Payment;
@Dao public interface PaymentDao {
 @Insert(onConflict=OnConflictStrategy.REPLACE) long insertPayment(Payment payment);
 @Query("SELECT * FROM payments WHERE appointmentId = :appointmentId LIMIT 1") Payment getPaymentByAppointment(int appointmentId);
 @Query("UPDATE payments SET status = :newStatus, paymentDate = :date WHERE paymentId = :id") void markAsPaid(int id,String newStatus,String date);
}

File: app/src/main/java/com/example/techfix/data/dao/PaymentDao.kt
package com.example.techfix.data.dao

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import com.example.techfix.data.entity.Payment

@Dao
interface PaymentDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insertPayment(payment: Payment): Long

    @Query("SELECT * FROM payments WHERE appointmentId = :appointmentId LIMIT 1")
    suspend fun getPaymentByAppointment(appointmentId: Int): Payment?

    @Query("UPDATE payments SET status = :newStatus, paymentDate = :date WHERE paymentId = :id")
    suspend fun markAsPaid(id: Int, newStatus: String, date: String)
}